mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -656,7 +656,8 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
         itemFactors.setName(s"itemFactors-$iter").persist(intermediateRDDStorageLevel)
         // TODO: Generalize PeriodicGraphCheckpointer and use it here.
         if (shouldCheckpoint(iter)) {
-          itemFactors.checkpoint() // itemFactors gets materialized in computeFactors.
+          // itemFactors gets materialized in computeFactors & here.
+          ALS.checkpointAndCleanParents(itemFactors)
```

Comment:
In the implicit case, `itemFactors` should be materialized in `computeYtY` (uses `aggregate`). It actually happens in L663 below (answering my previous question about why the checkpoint file deletion occurs after `userFactors` is computed.